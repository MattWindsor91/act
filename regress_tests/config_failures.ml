(* The Automagic Compiler Tormentor

   Copyright (c) 2018--2019 Matt Windsor and contributors

   ACT itself is licensed under the MIT License. See the LICENSE file in the
   project root for more information.

   ACT is based in part on code from the Herdtools7 project
   (https://github.com/herd/herdtools7) : see the LICENSE.herd file in the
   project root for more information. *)

(** Tests the output of the 'act c delitmus' command. *)

open Core

let parse_config_failures ~(file : Fpath.t) ~(path : Fpath.t) :
    unit Or_error.t =
  ignore file ;
  let r = Act_config.Frontend.load (Plumbing.Input.of_fpath path) in
  Fmt.(pr "@[%a@]@." (result ~ok:(always "No failures.") ~error:Error.pp)) r ;
  Ok ()

let run (test_dir : Fpath.t) : unit Or_error.t =
  let full_dir = Fpath.(test_dir / "config" / "failures" / "") in
  Common.regress_on_files "Config parser (failures)" ~dir:full_dir
    ~ext:"conf" ~f:parse_config_failures

let command =
  Common.make_regress_command
    ~summary:"runs config parser failure regressions" run

let () = Command.run command
