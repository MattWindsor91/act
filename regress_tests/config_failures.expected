# Config parser (failures) tests

## fuzz_empty_action.conf

```
("While reading from file:" input/config/failures/fuzz_empty_action.conf
 ("Parse error"
  (position input/config/failures/fuzz_empty_action.conf:2.9-3.1)
  (details "Expected an action identifier after `action`\n")))
```
## fuzz_empty_weight.conf

```
("While reading from file:" input/config/failures/fuzz_empty_weight.conf
 ("Parse error"
  (position input/config/failures/fuzz_empty_weight.conf:2.33-3.1)
  (details "Expected an integer weight after `weight`\n")))
```
## unknown_block.conf

```
("While reading from file:" input/config/failures/unknown_block.conf
 ("Parse error" (position input/config/failures/unknown_block.conf:1.1-1.5)
  (details
    "Expected a valid top-level item; for example:\
   \n- `herd { ... }`\
   \n- `machine ID { ... }`\
   \n- `compiler ID { ... }`\
   \n")))
```

Ran 3 test(s).
